src/commonMain/kotlin/com/github/quillraven/fleks/world.kt

```diff
@@ -326,8 +278,55 @@ class World internal constructor(
     /**
      * Returns the world's systems.
      */
-    var systems = emptyArray<IntervalSystem>()
-        internal set
+    private var _systems = mutableListOf<IntervalSystem>()
+    var systems: List<IntervalSystem>
+        get() = _systems
+        internal set(value) {
+            _systems = value.toMutableList()
+        }
+
+    /**
+     * Adds a new system to the world.
+     *
+     * @param system The system to be added to the world. This should be an instance of a class that extends IntervalSystem.
+     * @param index The position at which the system should be inserted in the list of systems. If null, the system is added at the end of the list.
+     * This parameter is optional and defaults to null.
+     *
+     * @throws FleksSystemAlreadyAddedException if the system was already added before.
+     */
+    fun add(system: IntervalSystem, index: Int? = null) {
```

Comment:
I would switch the argument because for normal list operations where you add an element at a specific index, the index parameter is first like in:

```kotlin
val numbers = mutableListOf("one", "five", "six")
numbers.add(1, "two")
```

To have a separate method without the index that simply adds at the end, I would go for a second method. To reduce code it could be something like this:

```kotlin
fun add(index: Int, system: IntervalSystem) {
  // do main logic here
}

fun add(system: IntervalSystem) = add(systems.size -1, system)
```